src/syngen/ml/worker/worker.py

```diff
@@ -301,6 +312,36 @@ def __train_tables(
                     both_keys=both_keys,
                     type_of_process=self.type_of_process,
                 )
+                MlflowTracker().end_run()
+
+        MlflowTracker().start_run(run_name='integral_metrics', tags={'process':'bottleneck'})
+        for table in chain_for_tables_for_training:
+            for stage in self.train_stages:
+                run_id = MlflowTracker().search_run(table, stage)
+                run_info = MlflowTracker().get_run(run_id).info
+                MlflowTracker().log_metric(key=f'{table}-{stage}-duration', value=(run_info.end_time - run_info.start_time)/1000)
+                for metric in self.hardware_metrics:
+                    try:
+                        MlflowTracker().log_metric(key=f'{table}-{stage}-{metric}', 
+                                                value=mean([m.value for m in MlflowTracker().get_metric_history(run_id, metric)])
+                                                )
+                    except:
+                        pass
+        if generation_of_reports:
```

Comment:
Does it make sense to move this logic to line 295?

Reply:
Sorry, hadn't got what do you mean